livekit-agents/livekit/agents/voice/agent_activity.py

```
@@ -1152,6 +1152,13 @@ async def _user_turn_completed_task(
             # is detected. So the previous execution should complete quickly.
             await old_task
 
+        if self._scheduling_paused:
```

`_scheduling_paused` is already checked before creating the `_user_turn_completed_atask` in `on_end_of_turn`.

---

I don't think this solves the problem, the activity can be still closed during `await self._agent.on_user_turn_completed` below.

---

I am wondering if that an expected raise when agent activity closed after that `on_end_of_turn`, or we should add a check right before the `_generate_reply` called. cc @theomonnom

---

we should probably raise an error on close, but still allow it when draining?